tests/python/relay/test_pass_dead_code_elimination.py

```diff
@@ -18,50 +18,56 @@
 
 import tvm
 from tvm import relay
-from tvm.relay.ir_pass import dead_code_elimination, alpha_equal
+from tvm.relay import Function, transform
+from tvm.relay.ir_pass import alpha_equal, graph_equal, free_vars
 from tvm.relay.op import log, add, equal, subtract
 
 
 class env:
     def __init__(self):
-        self.a = relay.Var("a")
-        self.b = relay.Var("b")
-        self.c = relay.Var("c")
-        self.d = relay.Var("d")
-        self.e = relay.Var("e")
-        self.x = relay.Var("x")
-        self.y = relay.Var("y")
-        self.z = relay.Var("z")
         self.shape = tvm.convert([1, 2, 3])
         self.tt = relay.TensorType(self.shape, "float32")
         self.int32 = relay.TensorType([], "int32")
         self.float32 = relay.TensorType([], "float32")
         self.one = relay.const(1.0)
         self.two = relay.const(2.0)
         self.three = relay.const(3.0)
+        self.a = relay.Var("a", self.float32)
+        self.b = relay.Var("b", self.float32)
+        self.c = relay.Var("c", self.float32)
+        self.d = relay.Var("d", self.float32)
+        self.e = relay.Var("e", self.float32)
+        self.x = relay.Var("x", self.int32)
+        self.y = relay.Var("y", self.int32)
+        self.z = relay.Var("z", self.int32)
 
 
 e = env()
 
 
 def test_let():
     orig = relay.Let(e.x, e.y, e.z)
-    assert alpha_equal(dead_code_elimination(orig), e.z)
+    orig = transform.OptimizeOnExpr(orig, transform.DeadCodeElimination())
+    assert alpha_equal(Function([orig], orig), Function([e.z], e.z))
 
 
 def test_used_let():
     orig = relay.Let(e.c, e.one, e.c + e.c)
-    assert alpha_equal(dead_code_elimination(orig), relay.Let(e.c, e.one, e.c + e.c))
+    orig = transform.OptimizeOnExpr(orig, transform.DeadCodeElimination())
+    expected = relay.Let(e.c, e.one, e.c + e.c)
+    assert alpha_equal(Function([e.c], orig), Function([e.c], expected))
 
 @nottest
 def test_inline():
     orig = relay.Let(e.a, e.b, relay.Let(e.c, e.d, e.c))
-    assert alpha_equal(dead_code_elimination(orig), e.d)
+    orig = transform.OptimizeOnExpr(orig, transform.DeadCodeElimination())
+    assert alpha_equal(Function([orig], orig), Function([e.d], e.d))
 
 
 def test_chain_unused_let():
     orig = relay.Let(e.a, e.b, relay.Let(e.c, e.d, e.e))
-    assert alpha_equal(dead_code_elimination(orig), e.e)
+    orig = transform.OptimizeOnExpr(orig, transform.DeadCodeElimination())
+    assert alpha_equal(Function([orig], orig), Function([e.e], e.e))
```

Comment:
this is not how function work.

Reply:
u can make a small function that take any expression, find it's free var, and wrap it in a function instead.

Reply:
I think this is actually okay because `orig` is a var in this case.

Reply:
> orig = relay.Let(e.a, e.b, relay.Let(e.c, e.d, e.e))

a free var mean a free variable, this is a let expression which is not a variable

Reply:
yeah, but it turns into a free variable after DCE. I can change it to free_vars(orig) if you think it is necessary.